src/macaron/slsa_analyzer/checks/check_result.py

```diff
@@ -24,6 +24,79 @@ class CheckResultType(str, Enum):
     UNKNOWN = "UNKNOWN"
 
 
+@dataclass
+class Evidence:
+    """The class representing an evidence generated by a check."""
+
+    #: The name of the evidence.
+    name: str
+
+    #: Determines whether the was evidence is found.
+    found: bool
+
+    #: The weight of the evidence.
+    weight: float
+
+
+class EvidenceWeightMap:
+    """This class creates a map object for collected evidence."""
+
+    def __init__(self, map_obj: dict[str, Evidence] | None = None) -> None:
+        """Initialize the class.
+
+        Parameters
+        ----------
+        map_obj: dict[str, Evidence] | None
+            The optional dictionary where keys are evidence names and values are the evidence objects.
+        """
+        self.map_obj = map_obj if map_obj else {}
+
+    def add(self, evidence: Evidence) -> None:
+        """Add an evidence to the map.
+
+        Parameters
+        ----------
+        evidence: Evidence
+            The evidence object.
+        """
+        self.map_obj[evidence.name] = evidence
+
+    def update_result(self, name: str, found: bool) -> None:
+        """Update the result if an evidence is found.
+
+        Parameters
+        ----------
+        found: bool
+            True if evidence was found.
+        """
+        if evidence := self.map_obj.get(name):
+            evidence.found = found
+
+    def get_max_score(self) -> float:
+        """Get the maximum possible score in this map.
+
+        Returns
+        -------
+        float
+            The maximum possible score or zero if the map is empty.
+        """
+        if self.map_obj.values():
+            return sum(e.weight for e in self.map_obj.values())
+        return 0
+
+    def get_score(self) -> float:
+        """Compute the score using the evidence result and weights.
+
+        Returns
+        -------
+        float
+            The aggregate score or zero if the map is empty.
+        """
+        if self.map_obj.values():
+            return sum(e.weight * int(e.found) for e in self.map_obj.values())
```

Comment:
nit: if the `map_obj` is an empty dictionary, wouldn't `sum(e.weight * int(e.found) for e in self.map_obj.values())` returns 0 anyway :thinking:

Reply:
See commit [6a06684](https://github.com/oracle/macaron/pull/673/commits/6a06684e73745585b65ecd83f7bc48f2ef2956a7).